CoreFoundation/String.subproj/CFString.c

```diff
@@ -5139,9 +5139,9 @@ void CFStringLowercase(CFMutableStringRef string, CFLocaleRef locale) {
 
     if (!langCode && isEightBit) {
         uint8_t *contents = (uint8_t *)__CFStrContents(string) + __CFStrSkipAnyLengthByte(string);
-        for (;currentIndex < length;currentIndex++) {
+        while (currentIndex < length) {
             if (contents[currentIndex] >= 'A' && contents[currentIndex] <= 'Z') {
-                contents[currentIndex] += 'a' - 'A';
+                contents[currentIndex++] += 'a' - 'A';
```

Comment:
I don't think this change should be merged. It is not completely equivalent in the case neither the `if` branch, nor the `else if` branch are taken, and will create an infinite loop.